src/scripts/main.js

```diff
@@ -1,3 +1,49 @@
 'use strict';
 
-// write your code here
+const promise1 = new Promise((resolve, reject) => {
+  const logo = document.querySelector('.logo');
+
+  if (logo) {
+    logo.addEventListener('click', () => {
+      resolve();
+    });
+  } else {
+    reject(new Error('Logo not found on the page'));
+  }
+});
+
+const promise2 = new Promise((resolve, reject) => {
+  setTimeout(() => {
+    reject(new Error('error'));
+  }, 3000);
+});
+
+const createMessage = (text, className = 'message') => {
+  const message = document.createElement('div');
+
+  message.classList.add(className);
+  message.textContent = text;
+
+  return message;
+};
+
+const insertMessageToDOM = (message) => {
+  document.body.appendChild(message);
+};
+
+const successHandler = () => {
+  const successMessage = createMessage('Promise was resolved!');
+
+  insertMessageToDOM(successMessage);
+};
+
+const errorHandler = (error) => {
+  const errormessage = createMessage(
+    `Promise was rejected! ${error.message}`,
+    'error-message',
+  );
+
+  insertMessageToDOM(errormessage);
+};
+
+Promise.race([promise1, promise2]).then(successHandler).catch(errorHandler);
```

Comment:
It's a good practice to handle errors in a `.catch` block after `Promise.race`. This will ensure that any errors that occur during the execution of the promises are properly handled.